src/main/scala/lms/thirdparty/cuda/cuda.scala

```diff
@@ -799,6 +814,68 @@ trait CudaOps extends Dsl with StackArrayOps with SizeTOps with CLibs with CudaF
       val i = blockIdxX * blockDimX * threadIdxX
       __ifThenElse(i < size, {y(i) = a * x(i) + y(i)}, {})
   }
+
+  def maskedFill[N:Numeric:Manifest](ijSwapped: Boolean)(implicit __pos: SourceContext) = cudaGlobalFun {
+    (in: Rep[Array[N]], out: Rep[Array[N]], mask: Rep[Array[Int]], value: Rep[N], 
+    dim0_shape: Rep[Int], dim1_shape: Rep[Int], dim0_stride: Rep[Int], dim1_stride: Rep[Int],
+    input_size: Rep[Int]) => {
+
```

Comment:
let's add some comments here to explain the parameters and how the caller should call it (what is the dim3 to use)